app/templating/summary/question.py

```diff
@@ -1,70 +1,73 @@
 import collections
 
-from app.data_model.answer_store import iterate_over_instance_ids
 from app.templating.summary.answer import Answer
 from app.questionnaire.rules import evaluate_skip_conditions
 
 
 class Question:
 
-    def __init__(self, question_schema, answers_map, answer_store, metadata):
+    def __init__(self, question_schema, answer_store, metadata):
         self.id = question_schema['id']
         self.type = question_schema['type']
         self.skip_conditions = question_schema.get('skip_conditions')
         answer_schema = question_schema['answers']
         self.title = question_schema['title'] or answer_schema[0]['label']
         self.number = question_schema.get('number', None)
-        self.answers = self._build_answers(question_schema, answer_schema, answers_map)
+        self._answers = answer_store.filter(escape_values=True)
+        self.answers_iterator = iter(answer_schema)
+        self.answers = self._build_answers(question_schema)
         self.is_skipped = evaluate_skip_conditions(self.skip_conditions, metadata, answer_store)
 
-    @classmethod
-    def _build_answers(cls, question_schema, answer_schema, answers):
+    def _get_answers(self, answer_id):
+        for answer_instance in self._answers:
+            if answer_instance['answer_id'] == answer_id:
+                yield answer_instance['value']
+
+    def _get_answer(self, answer_id):
+        try:
+            return next(self._get_answers(answer_id))
+        except StopIteration:
+            return None
+
+    def _build_answers(self, question_schema):
         summary_answers = []
-        answers_iterator = iter(answer_schema)
-        for answer_schema in answers_iterator:  # pylint: disable=redefined-argument-from-local
+        for answer_schema in self.answers_iterator:
             if 'parent_answer_id' in answer_schema:
                 continue
             if question_schema['type'] == 'RepeatingAnswer':
-                for answer_id, answer_index in iterate_over_instance_ids(answers.keys()):
-                    if answer_schema['id'] == answer_id:
-                        key = answer_id if answer_index == 0 else '_'.join([answer_id, str(answer_index)])
-                        answer = cls._build_answer(question_schema, answer_schema, answers, answers_iterator, key)
-                        summary_answers.append(Answer(answer_schema, answer))
+                for answer_value in self._get_answers(answer_schema['id']):
```

Comment:
The answer store already provides getting answers by id via the filter method - is there any reason to not use `answer_store.filter(answer_id= answer_schema['id'])`?

Reply:
@ajmaddaford keeping the helper methods in there but I agree that it was overkill rewriting a filter when one exists already.